torch/csrc/jit/tensorexpr/bounds_inference.cpp

```diff
@@ -128,36 +192,49 @@ static const Expr* simplifyMax(const Expr* a, const Expr* b) {
  * produce:
  *    [{a, kLoad, 0, 110}, {b, kStore, 0, 100}]
  */
-static BoundsInfo mergeTensorAccesses(const BoundsInfo& unmerged) {
+BoundsInfo mergeTensorAccesses(const BoundsInfo& unmerged) {
   BoundsInfo res;
-  std::unordered_map<const Buf*, TensorAccessBoundsInfo> merged;
-  for (const auto& t : unmerged) {
-    if (!merged.count(t.buf)) {
-      merged[t.buf] = t;
-      continue;
-    }
-
-    // We already have some range for this buf, try to merge them
-    TensorAccessBoundsInfo old_t = merged.at(t.buf);
-    TensorAccessBoundsInfo new_t = t;
-
-    for (size_t i = 0; i < old_t.start.size(); i++) {
-      new_t.start[i] = simplifyMin(old_t.start[i], new_t.start[i]);
-    }
-    for (size_t i = 0; i < old_t.stop.size(); i++) {
-      new_t.stop[i] = simplifyMax(old_t.stop[i], new_t.stop[i]);
+  for (auto& pair : unmerged) {
+    std::vector<TensorAccessBoundsInfo>& res_vec = res[pair.first];
+    for (const auto& v : pair.second) {
+      bool found = false;
+      for (auto& v2 : res_vec) {
```

Comment:
Do I understand it correctly that now the algorithm has become quadratic? Not that I expect it to be a problem, but want to double check whether I read the code correctly.

Reply:
Yeah it's n^2, specifically (n^2-n)/2 in worst case. I think n, which is the number of unmerged accesses to a specific buffer, will be small enough for this to be ok.